### ISSUE TITLE
Transaction succeeds with a revert error

### ISSUE THREAD

**Issue by petertonysmith94** [FuelLabs/fuels-ts#2898]
### Summary

A number of reports ([1](https://forum.fuel.network/t/ts-sdk-returns-an-error-even-though-tx-executed-successfully/6012), [2](https://forum.fuel.network/t/the-transaction-reverted-with-an-unknown-reason/6013)) of a transaction succeeding, but the SDK throwing a revert error:

```console
FuelError: The transaction reverted with an unknown reason
```

After investigations locally, I was unable to replicate this issue. However, using testnet, I was able to replicate this issue in around 1/10 transactions.

The issue arises due to a race condition, this occurs when:

- The [`statusChange`](https://github.com/FuelLabs/fuels-ts/blob/4e82ad42b84e520c3133907aabeea2aad1c1a199/packages/account/src/providers/transaction-response/transaction-response.ts#L225-L227) subscription can return a status of `SuccessStatus`
- The subsequencial [`getTransactionWithReceipts`](https://github.com/FuelLabs/fuels-ts/blob/4e82ad42b84e520c3133907aabeea2aad1c1a199/packages/account/src/providers/transaction-response/transaction-response.ts#L131-L133) query _can_ return with a status of `SubmittedStatus`

This leads to the transaction with `SubmittedStatus` being used as the transaction with no receipts and the following error and stack trace:
```console
_FuelError: The transaction reverted with an unknown reason.
    at assembleRevertError (/fuel/fuels-ts/packages/account/src/providers/utils/extract-tx-error.ts:109:10)
    at Module.extractTxError (/fuel/fuels-ts/packages/account/src/providers/utils/extract-tx-error.ts:142:10)
    at decodeCallResult (/fuel/fuels-ts/packages/program/src/script-request.ts:111:13)
    at decodeContractCallScriptResult (/fuel/fuels-ts/packages/program/src/contract-call-script.ts:145:20)
    at extractInvocationResult (/fuel/fuels-ts/packages/program/src/response.ts:29:26)
    at buildFunctionResult (/fuel/fuels-ts/packages/program/src/response.ts:64:17)
    at processTicksAndRejections (node:internal/process/task_queues:95:5)
    at createPool (/fuel/fuels-ts/packages/fuel-gauge/src/griffy-contract.test.ts:95:19)
    at test.repeats (/fuel/fuels-ts/packages/fuel-gauge/src/griffy-contract.test.ts:151:9)
    at runTest (file:///fuel/fuels-ts/node_modules/.pnpm/@vitest+runner@1.6.0/node_modules/@vitest/runner/dist/index.js:781:11) {
  VERSIONS: { FORC: '0.62.0', FUEL_CORE: '0.31.0', FUELS: '0.93.0' },
  metadata: { logs: [], receipts: [], panic: false, revert: false, reason: '' },
  rawError: {},
  code: 'script-reverted'
}
```

**Comment by arboleya** [comment]
Thanks for getting to the bottom of this. Seriously.

**Comment by petertonysmith94** [comment]
Relates to:
- https://github.com/FuelLabs/fuels-ts/issues/1644

**Comment by petertonysmith94** [comment]
We're currently exploring a solution at the network level.

**Comment by nedsalk** [comment]
This was resolved in #2896. #1644 will tackle the same problem but by using subscriptions instead of polling.
